File: 240201/이동경로상에 있는 모든 숫자 더하기/add-all-the-numbers-on-the-path.py
n,t=map(int,input().split())
dire=list(input())
grid=[list(map(int,input().split())) for _ in range(n)]

def search(n):
    x=y=(1+n)//2-1
    return x,y

def in_range(x,y):
    return 0<=x<n and 0<=y<n

dx,dy=[-1,0,1,0],[0,1,0,-1]
x,y=search(n)

ans=[grid[x][y]]
cur_dir=0
for cmd in dire:
    if cmd=='R':
        cur_dir=(cur_dir+1)%4
    elif cmd=='L':
        cur_dir=(cur_dir-1+4)%4
    else:
        nx,ny=x+dx[cur_dir],y+dy[cur_dir]
        if not in_range(nx,ny):
            continue
        x+=dx[cur_dir]
        y+=dy[cur_dir]
        ans.append(grid[x][y])


print(sum(ans))

File: 260104/이동경로상에 있는 모든 숫자 더하기/add-all-the-numbers-on-the-path.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');
const [n, t] = input[0].split(' ').map(Number);
const commands = input[1].split('');
const grid = input.slice(2).map((nums)=>nums.split(' ').map(Number));

const dx = [0, 1, 0, -1], dy = [1, 0, -1, 0];

function isRange(x, y){
    return 0<=x && x<n && 0<=y && y<n;
}

function getDir(cmd, dir){
    if(cmd==='R'){
        return (dir+1)%4;
    }else{
        return (dir-1+4)%4;
    }
}


const center = (n-1)/2;
let [x, y, dir] = [center, center, 3];

let ans = grid[x][y];

for(let cmd of commands){

    if(cmd === 'F'){
        const nx = x+dx[dir], ny = y+dy[dir];
        if(isRange(nx, ny)){
            x=nx, y=ny;
            ans+=grid[x][y];
        }
    }else{
        dir = getDir(cmd, dir);
    }
}

console.log(ans)